### ISSUE TITLE
How use a log scale with a kdeplot?

### ISSUE THREAD

**Issue by tlnagy** [mwaskom/seaborn#731]
```
import numpy as np, seaborn as sns, matplotlib.pyplot as plt
np.random.seed(1)
data = np.power(np.random.randn(1000), 10)
sns.kdeplot(data, shade=True)
plt.xscale('log')
plt.show()
```

looks pretty atrocious

![test](https://cloud.githubusercontent.com/assets/1661487/10418204/1adfadac-7009-11e5-943d-d953c5c108c7.png)

is the recommend solution just taking the log of `data` prior to plotting and then fixing the ticks? Or is there a better way?

**Comment by mwaskom** [comment]
> Is the recommend solution just taking the log of data prior to plotting and then fixing the ticks?

Yep; sorry.

**Comment by tlnagy** [comment]
How would you recommend to fix the ticks?

**Comment by mwaskom** [comment]
`ax.set(xticklabels=)`

**Comment by MaxGhenis** [comment]
Extracting the labels and transforming them turns out to be a bit complicated, since matplotlib [doesn't assign labels](https://stackoverflow.com/questions/41122923/getting-empty-tick-labels-before-showing-a-plot-in-matplotlib) before drawing them, and it gives a [medium dash](https://stackoverflow.com/questions/15538099/conversion-of-unicode-minus-sign-from-matplotlib-ticklabels) for negatives that breaks `int` conversion.

Here's one way:
```
import numpy as np, seaborn as sns, matplotlib.pyplot as plt
np.random.seed(1)
data = np.power(np.random.randn(1000), 10)

fig, ax = plt.subplots()
sns.kdeplot(np.log10(data), shade=True, ax=ax)
fig.canvas.draw()
locs, labels = plt.xticks()
ax.set(xticklabels=[10 ** int(i.get_text().replace(u'\u2212', '-'))
                    for i in labels])

plt.show()
```
![chart](https://user-images.githubusercontent.com/6076111/55990558-cc8c3c80-5c5c-11e9-82a1-5dd1e42fe9ae.png)

**Comment by mwaskom** [comment]
Closed in #2104 with the `log_scale` parameter in `kdeplot`.
